sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -181,6 +182,13 @@ case class BroadcastExchangeExec(
 }
 
 object BroadcastExchangeExec {
+  // Since the maximum number of keys that BytesToBytesMap supports is 1 << 29,
+  // and only 70% of the slots can be used before growing in HashedRelation,
+  // here the limitation should not be over 341 million.
+  val MAX_BROADCAST_TABLE_ROWS = ((1L << 29) / 1.5).toLong
```

Comment:
nit: `1L << 29` -> `(BytesToBytesMap.MAX_CAPACITY / 1.5).toLong`? Also, how about moving these constants into the `HashRelation` object since it seems these constants (e.g., 1.5) come from `HashRelation`?

Reply:
`val MAX_BROADCAST_TABLE_ROWS = (BytesToBytesMap.MAX_CAPACITY / 1.5).toLong`
Symbol MAX_CAPACITY is inaccessible from this place

Reply:
Yea, I think its ok to make it public.

Reply:
By counting in its semantic, I think we can keep `MAX_BROADCAST_TABLE_ROWS` here instead of `HashRelation`. The comment above is enough to explain why `/ 1.5`. Besides, we still have a constant `MAX_BROADCAST_TABLE_BYTES` here.